homeassistant/components/roomba/__init__.py

```diff
@@ -147,10 +148,16 @@ async def async_connect_or_timeout(hass, roomba):
 
 async def async_disconnect_or_timeout(hass, roomba):
     """Disconnect to vacuum."""
-    _LOGGER.debug("Disconnect vacuum")
-    with async_timeout.timeout(3):
-        await hass.async_add_job(roomba.disconnect)
-    return True
+    try:
+        with async_timeout.timeout(3):
+            _LOGGER.debug("Disconnect vacuum")
+            await hass.async_add_executor_job(roomba.disconnect)
+            while roomba.roomba_connected:
+                await asyncio.sleep(1)
+                break
```

Comment:
I think the problem here is that you are using while loop and break in the first iteration. You should either use if statement if you just want to sleep for 1 second if it is still connected, or you can remove the `break` so the while loop will wait until it's disconnected, or you just remove the loop and it will wait for 1 second anyway. If I understand your comment correctly, you want to wait until disconnected and then wait for another second to make sure the reconnect works well. Then you'll need to delete the `break` statement and add another `sleep` outside the while loop.